--- modulesets/gtk-osx.modules
+++ modulesets/gtk-osx.modules
@@ -109,8 +109,9 @@
        path so none of the libraries that depend on it can link. Require
        the install name to be an absolute path.
   -->
+  <!-- gpodder github action needs to disable BrotliDec -->
   <cmake id="freetype-no-harfbuzz"
-         cmakeargs='-DFT_DISABLE_HARFBUZZ=ON -DBUILD_SHARED_LIBS=true -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_NAME_DIR="${JHBUILD_PREFIX}/lib"'>
+         cmakeargs='-DFT_DISABLE_HARFBUZZ=ON -DBUILD_SHARED_LIBS=true -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_NAME_DIR="${JHBUILD_PREFIX}/lib" -DCMAKE_DISABLE_FIND_PACKAGE_BrotliDec=TRUE'>
     <branch module="freetype/freetype-2.13.3.tar.xz"
             version="2.13.3"
             hash="sha256:0550350666d427c74daeb85d5ac7bb353acba5f76956395995311a9c6f063289"
@@ -141,8 +141,9 @@
       <dep package="icu" />
     </dependencies>
   </meson>
+   <!-- gpodder github action needs to disable BrotliDec -->
   <cmake id="freetype"
-         cmakeargs='-DFT_REQUIRE_HARFBUZZ=ON -DBUILD_SHARED_LIBS=true -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_NAME_DIR="${JHBUILD_PREFIX}/lib"'>
+         cmakeargs='-DFT_REQUIRE_HARFBUZZ=ON -DBUILD_SHARED_LIBS=true -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_NAME_DIR="${JHBUILD_PREFIX}/lib" -DCMAKE_DISABLE_FIND_PACKAGE_BrotliDec=TRUE'>
     <branch module="freetype/freetype-2.13.3.tar.xz"
             version="2.13.3"
             hash="sha256:0550350666d427c74daeb85d5ac7bb353acba5f76956395995311a9c6f063289"
