--- modulesets/gtk-osx.modules
+++ modulesets/gtk-osx.modules
@@ -127,8 +127,9 @@
     then build harfbuzz without cairo because cairo requires
     harfbuzz.
   -->
+  <!-- gpodder github action needs to disable BrotliDec -->
   <cmake id="freetype-no-harfbuzz"
-         cmakeargs="-DCMAKE_DISABLE_FIND_PACKAGE_HarfBuzz=TRUE -DCMAKE_DISABLE_FIND_PACKAGE_BZip2=TRUE -D BUILD_SHARED_LIBS=true -D CMAKE_BUILD_TYPE=Release">
+         cmakeargs="-DCMAKE_DISABLE_FIND_PACKAGE_HarfBuzz=TRUE -DCMAKE_DISABLE_FIND_PACKAGE_BZip2=TRUE -D BUILD_SHARED_LIBS=true -D CMAKE_BUILD_TYPE=Release -DCMAKE_DISABLE_FIND_PACKAGE_BrotliDec=TRUE">
 
     <branch module="freetype/freetype-2.11.1.tar.gz"
             version="2.11.1"
@@ -153,8 +154,9 @@
     </dependencies>
   </meson>
   <!---->
+   <!-- gpodder github action needs to disable BrotliDec -->
   <cmake id="freetype"
-         cmakeargs="-DCMAKE_DISABLE_FIND_PACKAGE_BZip2=TRUE -D BUILD_SHARED_LIBS=true -D CMAKE_BUILD_TYPE=Release">
+         cmakeargs="-DCMAKE_DISABLE_FIND_PACKAGE_BZip2=TRUE -D BUILD_SHARED_LIBS=true -D CMAKE_BUILD_TYPE=Release -DCMAKE_DISABLE_FIND_PACKAGE_BrotliDec=TRUE">
 
     <branch module="freetype/freetype-2.11.1.tar.gz"
             version="2.11.1"
