--- src/gpodder/gtkui/main.py.orig	2011-11-08 20:25:04.000000000 +0100
+++ src/gpodder/gtkui/main.py	2011-11-08 20:30:18.000000000 +0100
@@ -35,6 +35,7 @@
 import urllib
 import cgi
 
+from gtk_osxapplication import *
 
 import gpodder
 
@@ -92,7 +93,7 @@
     # Width (in pixels) of episode list icon
     EPISODE_LIST_ICON_WIDTH = 40
 
-    def __init__(self, bus_name, gpodder_core):
+    def __init__(self, bus_name, gpodder_core, macapp):
         dbus.service.Object.__init__(self, object_path=gpodder.dbus_gui_object_path, bus_name=bus_name)
         self.podcasts_proxy = DBusPodcastsProxy(lambda: self.channels, \
                 self.on_itemUpdate_activate, \
@@ -104,6 +105,7 @@
         self.config = self.core.config
         self.db = self.core.db
         self.model = self.core.model
+        self.macapp = macapp
         BuilderWidget.__init__(self, None)
     
     def new(self):
@@ -134,28 +136,20 @@
 
         # Mac OS X-specific UI tweaks: Native main menu integration
         # http://sourceforge.net/apps/trac/gtk-osx/wiki/Integrate
-        if getattr(gtk.gdk, 'WINDOWING', 'x11') == 'quartz':
-            try:
-                import igemacintegration as igemi
 
-                # Move the menu bar from the window to the Mac menu bar
-                self.mainMenu.hide()
-                igemi.ige_mac_menu_set_menu_bar(self.mainMenu)
-
-                # Reparent some items to the "Application" menu
-                for widget in ('/mainMenu/menuHelp/itemAbout', \
-                               '/mainMenu/menuPodcasts/itemPreferences'):
-                    item = self.uimanager1.get_widget(widget)
-                    group = igemi.ige_mac_menu_add_app_menu_group()
-                    igemi.ige_mac_menu_add_app_menu_item(group, item, None)
-
-                quit_widget = '/mainMenu/menuPodcasts/itemQuit'
-                quit_item = self.uimanager1.get_widget(quit_widget)
-                igemi.ige_mac_menu_set_quit_menu_item(quit_item)
-            except ImportError:
-                print >>sys.stderr, """
-                Warning: ige-mac-integration not found - no native menus.
-                """
+        # Move the menu bar from the window to the Mac menu bar
+        self.mainMenu.hide()
+        self.macapp.set_menu_bar(self.mainMenu)
+
+        # Reparent some items to the "Application" menu
+        item = self.uimanager1.get_widget('/mainMenu/menuHelp/itemAbout')
+        self.macapp.insert_app_menu_item(item, 0)
+        self.macapp.insert_app_menu_item(gtk.SeparatorMenuItem(), 1)
+        item = self.uimanager1.get_widget('/mainMenu/menuPodcasts/itemPreferences')
+        self.macapp.insert_app_menu_item(item, 2)
+
+        quit_item = self.uimanager1.get_widget('/mainMenu/menuPodcasts/itemQuit')
+        quit_item.hide()
 
         self.download_status_model = DownloadStatusModel()
         self.download_queue_manager = download.DownloadQueueManager(self.config)
@@ -2443,6 +2437,13 @@
             self.close_gpodder()
 
         return True
+       
+    def quit_cb(self, macapp):
+        """Called when OSX wants to quit the app (Cmd-Q or gPodder > Quit)
+        """
+        # event can't really be cancelled : crash when it's done
+        self.close_gpodder()
+        return False
 
     def close_gpodder(self):
         """ clean everything and exit properly
@@ -2458,7 +2459,6 @@
         self.core.shutdown()
 
         self.quit()
-        sys.exit(0)
 
     def get_expired_episodes(self):
         for channel in self.channels:
@@ -3352,7 +3352,10 @@
         dlg.destroy()
         sys.exit(0)
 
-    gp = gPodder(bus_name, core.Core(UIConfig, model_class=Model))
+    macapp = OSXApplication()
+
+
+    gp = gPodder(bus_name, core.Core(UIConfig, model_class=Model), macapp)
 
     # Handle options
     if options.subscribe:
@@ -3363,6 +3366,8 @@
     if platform.system() == 'Darwin':
         from gpodder.gtkui import macosx
         macosx.register_handlers(gp)
+        macapp.connect("NSApplicationBlockTermination", gp.quit_cb)
+        macapp.ready()
     # end mac OS X stuff
 
     gp.run()
--- src/gpodder/util.py.orig	2011-11-08 20:34:52.000000000 +0100
+++ src/gpodder/util.py	2011-11-08 20:35:30.000000000 +0100
@@ -1144,6 +1144,8 @@
     try:
         if gpodder.win32:
             os.startfile(filename)
+        elif platform.system() == 'Darwin':
+            subprocess.Popen(['open', filename])
         else:
             subprocess.Popen(['xdg-open', filename])
         return True
