--- bin/gpodder.orig	2016-04-26 08:51:22.850002163 +0200
+++ bin/gpodder	2016-04-26 08:53:47.816670409 +0200
@@ -144,10 +144,6 @@
             except dbus.exceptions.DBusException, dbus_exception:
                 logger.info('Cannot connect to remote object.', exc_info=True)
 
-    if not gpodder.ui.win32 and os.environ.get('DISPLAY', '') == '':
-        logger.error('Cannot start gPodder: $DISPLAY is not set.')
-        sys.exit(1)
-
     if gpodder.ui.gtk:
         from gpodder.gtkui import main
         gpodder.ui_folders.insert(0, os.path.join(ui_folder, 'gtk'))
