--- configure.old	2020-07-20 15:01:32.000000000 +0200
+++ configure	2021-05-27 16:43:37.000000000 +0200
@@ -11506,7 +11506,7 @@
  clock confstr copy_file_range ctermid dup3 execv explicit_bzero explicit_memset \
  faccessat fchmod fchmodat fchown fchownat \
  fdwalk fexecve fdopendir fork fpathconf fstatat ftime ftruncate futimesat \
- futimens futimes gai_strerror getentropy \
+ futimens futimes gai_strerror \
  getgrgid_r getgrnam_r \
  getgrouplist getgroups getlogin getloadavg getpeername getpgid getpid \
  getpriority getresuid getresgid getpwent getpwnam_r getpwuid_r getspnam getspent getsid getwd \
@@ -12924,7 +12924,7 @@
 
 for ac_func in clock_gettime
 do :
-  ac_fn_c_check_func "$LINENO" "clock_gettime" "ac_cv_func_clock_gettime"
+  ac_fn_c_check_func "$LINENO" "clock_gettimexx" "ac_cv_func_clock_gettime"
 if test "x$ac_cv_func_clock_gettime" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_CLOCK_GETTIME 1
@@ -12986,7 +12986,7 @@
 
 for ac_func in clock_getres
 do :
-  ac_fn_c_check_func "$LINENO" "clock_getres" "ac_cv_func_clock_getres"
+  ac_fn_c_check_func "$LINENO" "clock_getresxx" "ac_cv_func_clock_getres"
 if test "x$ac_cv_func_clock_getres" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_CLOCK_GETRES 1
@@ -13044,7 +13044,7 @@
 
 for ac_func in clock_settime
 do :
-  ac_fn_c_check_func "$LINENO" "clock_settime" "ac_cv_func_clock_settime"
+  ac_fn_c_check_func "$LINENO" "clock_settimexx" "ac_cv_func_clock_settime"
 if test "x$ac_cv_func_clock_settime" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_CLOCK_SETTIME 1
