--- configure.ac.old	2015-12-23 22:35:52.000000000 +0100
+++ configure.ac	2015-12-23 22:36:03.000000000 +0100
@@ -409,10 +409,6 @@
 if test "$with_unicode_backend" = "icu"; then
     # TODO: use pkg-config (after CFLAGS in their .pc files are cleaned up)
     case "$host" in
-        *-*-darwin*)
-            UNICODE_CFLAGS="-I$srcdir/Source/JavaScriptCore/icu -I$srcdir/Source/WebCore/icu"
-            UNICODE_LIBS="-licucore"
-            ;;
         *-*-mingw*)
             UNICODE_CFLAGS=""
             UNICODE_LIBS="-licui18n -licuuc"
