--- setup.py.orig	2016-06-09 21:20:28.000000000 +0200
+++ setup.py	2016-06-09 21:22:56.000000000 +0200
@@ -792,18 +792,13 @@
                               '/usr/local/ssl/include',
                               '/usr/contrib/ssl/include/'
                              ]
-        ssl_incs = find_file('openssl/ssl.h', inc_dirs,
-                             search_for_ssl_incs_in
-                             )
+        ssl_incs = [os.environ["C_INCLUDE_PATH"]]
         if ssl_incs is not None:
             krb5_h = find_file('krb5.h', inc_dirs,
                                ['/usr/kerberos/include'])
             if krb5_h:
                 ssl_incs += krb5_h
-        ssl_libs = find_library_file(self.compiler, 'ssl',lib_dirs,
-                                     ['/usr/local/ssl/lib',
-                                      '/usr/contrib/ssl/lib/'
-                                     ] )
+        ssl_libs = [os.environ["JHBUILD_LIBDIR"]]
 
         if (ssl_incs is not None and
             ssl_libs is not None):
